lib/std/fmt.zig

```diff
@@ -444,6 +444,15 @@ fn defaultSpec(comptime T: type) [:0]const u8 {
     return "";
 }
 
+// This errors when we're given an explicit type specifier (i.e. not "any" or nothing).
+fn checkNoSpecifier(comptime thing: []const u8, comptime actual_fmt: []const u8) void {
+    if (actual_fmt.len != 0)
+        @compileError(comptime comptimePrint(
+            \\cannot format {s} with specifier '{s}'
+            \\try removing the specifier
+        , .{ thing, actual_fmt }));
```

Comment:
This should match other similar errors so:
```zig
"Unsupported format string '" ++ fmt ++ "' for type '" ++ @typeName(@TypeOf(value)) ++ "'"
```

Reply:
Are you talking about the formatted message or the code here?
Is it bad to use `comptimePrint` here because I think it's much more readable than those long `++` one-liners and it did not cause any problems for me when I tested this error with `comptimePrint` itself.
I do agree that the usage of a newline here is at least currently inconsistent with all the other messages because people just aren't used to the fact that they can use multiline strings now and they'll look pretty. `comptimePrint` doesn't seem to cause problems either. But that's just my style.
So if you insist, I can change it of course, but I would be interested in your opinion on this.

I chose to give the user a hint here (`"try removing the specifier"`) because here I'm in a pretty good position to say that removing the specifier will probably fix your problem. However looking at the mistakes I found with this, removing the specifier wasn't always the solution, so maybe it's still better to just remove it... yeah I will just do that.

If you're talking about the formatted string itself, that was kind of intentional because to be honest I never quite agreed with capitalizing a `@compileError` message because usually all errors are entirely in lowercase and the compiler does it that way too. It would only be consistent.
And we already have such messages elsewhere anyway: https://github.com/ziglang/zig/blob/0df702d81a4d5e6b478a44b4037c5b62c6c2a2d7/lib/std/fmt.zig#L587

Reply:
I was talking about the content of the message, I think it is bad to have multiple different messages for the same kind of error, `comptimePrint` or `++` are either fine though `++` is much simpler.

I also prefer compile errors to be lowercased, I just directly copied the error found elsewhere, it would be better for those other uses to also be lowercase, or better yet, call this new function.

Reply:
Ah yes, I understand now. You're right! I refactored it and made it into a more general function to be used for such errors.

Reply:
@Vexu while I'm at it can I push a commit to this PR that lowercases all compile errors in just this file?

Reply:
Fine by me.